metrics.go

```diff
@@ -15,55 +15,105 @@ type CertificateDesc struct {
 }
 
 type MetricsCollector struct {
-	certsMetric *prometheus.Desc
+	certsMetric  *prometheus.Desc
+	errorsMetric *prometheus.Desc
 
-	role           string
-	updates        chan VaultEvent
-	resourceEvents map[string]VaultEvent
-	resourcesMutex sync.RWMutex
+	role string
+
+	resourcesUpdates chan VaultEvent
+	resourceEvents   map[string]VaultEvent
+
+	errorUpdates  chan string
+	errorCounters map[string]int
+
+	metricsMutex sync.RWMutex
+}
+
+var errorUpdates chan string
+var errorUpdatesMutex sync.Mutex
+
+func setErrorUpdatesChan(ch chan string) {
+	errorUpdatesMutex.Lock()
+	defer errorUpdatesMutex.Unlock()
+
+	errorUpdates = ch
+}
+
+func getErrorUpdatesChan() chan string {
+	errorUpdatesMutex.Lock()
+	defer errorUpdatesMutex.Unlock()
+
+	return errorUpdates
 }
 
 func (m MetricsCollector) init() {
 	for {
 		select {
-		case event := <-m.updates:
+		case event := <-m.resourcesUpdates:
 			if event.Type == EventTypeFailure {
 				continue
 				// TODO: emit some nice metrics here
 			}
 
 			id := fmt.Sprintf("%v:%v", event.Resource.Resource, event.Resource.Path)
-			m.resourcesMutex.Lock()
+			m.metricsMutex.Lock()
 			m.resourceEvents[id] = event
-			m.resourcesMutex.Unlock()
+			m.metricsMutex.Unlock()
+		case err := <-m.errorUpdates:
+			m.metricsMutex.Lock()
+			m.errorCounters[err]++
+			m.metricsMutex.Unlock()
 		}
 	}
 }
 
-func RegisterMetricsCollector(role string, updates chan VaultEvent) {
+func IncrementErrorCounters(label string) {
```

Comment:
If we do choose to make this a method on the `MetricsCollector` struct, perhaps this could just named `Error(label string)`?